### ISSUE TITLE
Problem parsing package & OTA Update Failure: 400

### ISSUE THREAD

**Issue by Braxwell** [4Q-s-r-o/ota_update#39]
Hi !
I have some problems implementing OTA update. I don't think it's a problem of link or file because I can manually install the app by downloading it.
When I try to update my app (using the same code as the example you gave), this happens (permissions have been granted) :

`D/FLUTTER OTA( 5335): OTA UPDATE ON LISTEN`
`D/FLUTTER OTA( 5335): OTA UPDATE ON STARTING DOWNLOAD`
`D/FLUTTER OTA( 5335): OTA UPDATE DOWNLOAD STARTED 4479`
`D/FLUTTER OTA( 5335): OTA UPDATE TRACK DOWNLOAD STARTED 4479`
`D/FLUTTER OTA( 5335): OTA UPDATE TRACK DOWNLOAD THREAD STARTED 4479`
`D/FLUTTER OTA( 5335): OTA UPDATE PENDING`
`D/FLUTTER OTA( 5335): OTA UPDATE FAILURE: 400`

And then I have the "There was a problem parsing the package." message on the screen. When I go to my download folder, there is no apk file. I tested with and without checksum, and with it I have the same results, minus the message on screen.
If it can help, I'm using a Xiaomi Mi A2 (Android 10) for my tests.

Thank you !

**Comment by MartinHlavna** [comment]
That number is just error reason provided from Android download manager. According to [documentation](https://developer.android.com/reference/android/app/DownloadManager#COLUMN_REASON) it will hold either ERROR_ constant or HTTP status code. All ERROR constant are above 1000 so it looks like your server responded with [HTTP 400](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400) status.

Could you see error logs on your server? Maybe that server requires some headers.

**Comment by Braxwell** [comment]
After some tests, It seems that my requests don't even reach the server. I switched my link for the one showed in the example and it worked perfectly.
In this application, I send another request for the same server (to a .php file) and it's working, so I find this strange. I also tried to download the file from another link (not a private server) and it didn't work.
Maybe OTA Update is blocking non-httpS addresses ?
